Ros2 devel

swatifulzele opened the pull request:
hello

vandanamandlik committed:
added setup.cfg

vandanamandlik committed:
Modifed package.xml according to ROS2 rules.
- Changed run_depend tag to exec_depend as per ROS2 rules.
- Added new tag build_type with value ament_python.
- Renamed roslaunch package name to ros2launch.

vandanamandlik committed:
Replaced distutils.core package by setuptools package.

rhaschke commented:
Thank you very much for this contribution. However, I'm not sure yet how to deal with this. I'm not very familar with ROS2 yet. But, obviously there are incompatible changes in the make system (catkin -> ament) as well as an incompatible package name change (roslaunch -> roslauch2).

rhaschke commented:
Dear @codebot @wjwwood,
the common way to handle ROS1 -> ROS2 transition is to clone the repo on github.com/ros2/xacro.
Could anybody with appropriate rights please do so and grant me maintainer status there as well?

The main issue for migrating to ROS2 seems to be the build system (`catkin` -> `ament`). However, I have read about [`catment`](https://github.com/ros2/ros2/wiki/catment) as well. According to this Wiki, it should be possible to stick with the catkin-based cmake file and package.xml in this package, and still successfully build with ament. What's the status of this catment thing? Is it well established practice or alpha testing only?

Is there a way to specify different package dependencies for ROS1 and ROS2 in `package.xml` (as required here for `ros2launch` opposed to `roslaunch`)?

vandanamandlik committed:
removed some packages those were not required eg. std_msgs, roslint etc.

vandanamandlik committed:
Modified setup.py to use python 3.5 version.
Added data_files configuration settings.
Modified entry_points setttings.

vandanamandlik committed:
Moved contents of scripts folder to resource folder.

vandanamandlik committed:
Moved contents from "src/xacro/" folder to "xacro/" folder.

vandanamandlik committed:
Added temporary workaround for rosgraph.names.load_mappings because rosgraph is not migrated to ROS2.

vandanamandlik committed:
Resolved errors where some test cases were failing to find out some .xml files. (eg. include1.xml)

vandanamandlik committed:
Updated shebang line to use python3 in resource/xacro and setup.py
removed unnecessary code from setup.py and some cosmetic changes as per ros2 guidelines.

vandanamandlik committed:
Added linter test cases- ament_copyright, ament_flake8 and ament_pep257

vandanamandlik committed:
Removed deprecated xacro.py file
Modified all python scripts as per ros2 coding guidelines.
Added substitution_args.py file, replacement for roslaunch's resolve_args() method,
as it is not available in ROS2's ros2launch package.

vandanamandlik committed:
Removed linter test cases

vandanamandlik committed:
Added linter test cases- ament_copyright, ament_flake8 and ament_pep257
Changed license from BSD to 'Apache License, Version 2.0'.(As per ROS2 rules.)
Modified code according to linter test rules
eg.
- Unnecessary dict call - rewrite as a literal
- Indentation error
- Remove bad quotes

vandanamandlik committed:
Added files CONTRIBUTING.md and LICENSE (Apache License, Version 2.0), required for ament_copyright rule.

vandanamandlik committed:
Removed CMakeLists.txt
As per ROS2 Migration guide If the ROS 1 package uses CMake only to invoke the setup.py file and does not contain anything beside Python code (e.g. also no messages, services, etc.) it should be converted into a pure Python package in ROS 2.
so removed the CMakeLists.txt file.
Referred link: https://index.ros.org/doc/ros2/Migration-Guide/

vandanamandlik committed:
Fixed following errors as per flake8 rules
- A001 "input" is a python builtin and is being shadowed, consider renaming the variable.
- A002 "str" is used as an argument and thus shadows a python builtin, consider renaming the argument.
-  A003 "next" is a python builtin, consider renaming the class attribute.

vandanamandlik committed:
Remove bad quotes as per flake8 rules.

vandanamandlik committed:
Resolved following cosmetic errors:
- Remove bad quotes
- Remove bad quotes from multiline string
- line break before binary operator
- line break after binary operator
- invalid escape sequence '\('
Added shebang line in test_copyright.py, test_flake8.py and test_pep257.py scripts to use python3.

vandanamandlik committed:
Removed .travis.yml for now, as it is ros1 setup file.
